### PR TITLE
[incubator-kie-drools-5808] [new-parser] Combining DRLParser.g4 and DRL6Expressions.g4

### PR DESCRIPTION
**Pull request by tkobayas**
**Issue**:

* https://github.com/apache/incubator-kie-drools/issues/5808

I raised this PR on top of @yurloc 's effort to import `DRL6Expressions.g4`, thanks! Now we removed the duplication around constraint expression parser rules. This PR doesn't yet implement Visitor pattern to `DRL6Expressions.g4`. It can be done some time later.

### Before PR in `drools-model-codegen`
```
[ERROR] Tests run: 2434, Failures: 140, Errors: 0, Skipped: 9
```

### After PR in `drools-model-codegen`
```
[ERROR] Tests run: 2434, Failures: 97, Errors: 0, Skipped: 9
```

<details>
<summary>
How to replicate CI configuration locally?
</summary>

Build Chain tool does "simple" maven build(s), the builds are just Maven commands, but because the repositories relates and depends on each other and any change in API or class method could affect several of those repositories there is a need to use [build-chain tool](https://github.com/kiegroup/github-action-build-chain) to handle cross repository builds and be sure that we always use latest version of the code for each repository.
 
[build-chain tool](https://github.com/kiegroup/github-action-build-chain) is a build tool which can be used on command line locally or in Github Actions workflow(s), in case you need to change multiple repositories and send multiple dependent pull requests related with a change you can easily reproduce the same build by executing it on Github hosted environment or locally in your development environment. See [local execution](https://github.com/kiegroup/github-action-build-chain#local-execution) details to get more information about it.
</details>

<details>
<summary>
How to retest this PR or trigger a specific build:
</summary>

- for <b>pull request and downstream checks</b>  
  - Push a new commit to the PR. An empty commit would be enough.

- for a <b>full downstream build</b>
  - for <b>github actions</b> job: add the label `run_fdb`

- for <b>Jenkins PR check only</b>
  - If you are an ASF committer for KIE podling, login to Jenkins (https://ci-builds.apache.org/job/KIE/job/drools/), go to the specific PR job, and click on `Build Now` button.
</details>

### CODE REVIEW

File: drools-drl/drools-drl-parser/src/main/antlr4/org/drools/drl/parser/antlr4/DRL6Expressions.g4 [hunk 1]
```diff
@@ -30,12 +30,7 @@ options {
 @members {
     private ParserHelper helper;
 
-    public DRL6Expressions(TokenStream input,
-                          ParserHelper helper ) {
-        this( input );
-        this.helper = helper;
-    }
-
+    public void setHelper( ParserHelper helper )              { this.helper = helper; }
```
Comment (tkobayas) [inline drools-drl/drools-drl-parser/src/main/antlr4/org/drools/drl/parser/antlr4/DRL6Expressions.g4:33]: Removing the `DRL6Expressions` constructor is okay. We can use another constructor `new DRL6Expressions(input)` and `setHelper`.  `DRLParser` can also inject `ParserHelper`.

File: drools-drl/drools-drl-parser/src/main/antlr4/org/drools/drl/parser/antlr4/DRLParser.g4 [hunk 2]
```diff
@@ -446,7 +311,7 @@ patternSource : fromAccumulate
               | fromExpression
               ;
 
-fromExpression : unaryExpressionNotPlusMinus ;
+fromExpression : conditionalOrExpression ;
```
Comment (tkobayas) [inline drools-drl/drools-drl-parser/src/main/antlr4/org/drools/drl/parser/antlr4/DRLParser.g4:314]: @yurloc  I have reverted several rules which you committed in https://github.com/apache/incubator-kie-drools/pull/5834  Is it okay? Let me know if I'm reverting too much.

File: drools-drl/drools-drl-parser/src/main/antlr4/org/drools/drl/parser/antlr4/DRL6Expressions.g4 (hunk 1 above)
Comment (tkobayas) [reply drools-drl/drools-drl-parser/src/main/antlr4/org/drools/drl/parser/antlr4/DRL6Expressions.g4:33]: There may be room to improve things around `@member` maybe in a separated PR.

**Comment by tkobayas** [comment]
@yurloc @mariofusco @gitgabrio Please review, thanks!

**Comment by tkobayas** [comment]
@yurloc if this PR introduces regressions in `drools-traits` tests, let me know. I'll investigate (but it might be a separated PR)

File: drools-drl/drools-drl-parser/src/main/antlr4/org/drools/drl/parser/antlr4/DRLParser.g4 (hunk 2 above)
Comment (yurloc) [reply drools-drl/drools-drl-parser/src/main/antlr4/org/drools/drl/parser/antlr4/DRLParser.g4:314]: Makes sense.

**Review by yurloc — APPROVED** [review]
Awesome PR, thanks @tkobayas!
- No traits regressions.
- Fixes several tests failing due to lack of BigInteger and BigDecimal support (known but not reported).
- Fixes https://github.com/apache/incubator-kie-drools/issues/5725.
- Fixes https://github.com/apache/incubator-kie-drools/issues/5739.

**Review by mariofusco — APPROVED** [review]

**Comment by yurloc** [comment]
Overall fixing progress: There were 224 failed tests.

**Pull request merged**